### PR TITLE
Add possibility to change styles of sortable elements, mainly to change colors and background colors.

### PR DESCRIPTION
**Pull request by Socvest**
Perhaps using [glamor](https://www.npmjs.com/package/glamor) so we can inject styles in-line?

We can then send in a dictionary of the classNames with the desired styles which will override SortableComponent.css file?

Using the streamlit's designs may not always be desireable.

### CODE REVIEW

File: streamlit_sortables/frontend/src/SortableItem.tsx [hunk 1]
```diff
@@ -28,7 +29,7 @@ export const SortableItem: FunctionComponent<SortableItemProps> = ((props) => {
     const className = "btn shadow-none sortable-item " + (props.isActive ? "active" : "");
 
     return (
-      <li className={className} ref={setNodeRef} style={style} {...attributes} {...listeners}>
+      <li className={className} {...style(props.inLineStyles[className])} ref={setNodeRef} style={style} {...attributes} {...listeners}>
```
Comment (ohtaman) [inline streamlit_sortables/frontend/src/SortableItem.tsx:32]: this does not work in my environment. `style` indicates the variable defined at l.24

**Review by ohtaman — COMMENTED** [review]
Thank you! Here are a few comments:

1. I think making it possible to apply CSS is a good idea.  
2. It didn’t work in my environment. Did it work in yours?  
3. I’m not very familiar with this, but is *glamor* necessary? (If simply providing a dictionary to `style` works, I think that would be preferable. I'm not sure the library is needed or not.)

File: streamlit_sortables/frontend/src/SortableItem.tsx (hunk 1 above)
Comment (Socvest) [reply streamlit_sortables/frontend/src/SortableItem.tsx]: Ah, my mistake, did not realise style was defined elsewhere there. Sorted via:

`import { style as StyleGlamor } from "glamor";`
`{...StyleGlamor(props.inLineStyles[className])}`

**Comment by Socvest** [comment]
> Thank you! Here are a few comments:

> 3. I’m not very familiar with this, but is _glamor_ necessary? (If simply providing a dictionary to `style` works, I think that would be preferable. I'm not sure the library is needed or not.)

We don't necessarily need to use glamor. We can pass a dictionary and input into the `style` variable in the tag element. Something like this can work

```

{
    "background-color": "red",
    height: "100px",
    width: "100px",
  }

```

**Comment by Socvest** [comment]
Let me sort some things out.

**Comment by Socvest** [comment]
Okay, try out the new changes I have made. It should work now.

Below is an example from this:

```
in_line_styles = {
        'sortable-component-parent-container':{
            "backgroundColor":"blue",
        },
        'sortable-item':{
            "backgroundColor":"brown",
        },
        'sortable-container':{
            'backgroundColor':"orange"
        }
    }

```

image

**Comment by ohtaman** [comment]
I have confirmed that it works as expected.
I'll release a version including this update within a few days.

Thanks for your contribution!

**Review by ohtaman — APPROVED** [review]

**Comment by ohtaman** [comment]
[FYI] I was very happy with your contribution, but I noticed that with the current implementation, the styles do not change on :hover (#21). Therefore, I am considering either of the following approaches:

1. Simply adding a <style> tag.
2. Making it possible to use the parent's style.

**Pull request merged**